src/components/common/DataTable.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <div class="table-responsive">
+    <table :class="darkMode ? 'dark-table' : 'table'">
+      <thead>
+        <tr :class="darkMode ? 'table-header-dark' : 'table-header-light'">
+          <th
+            @click="setSortBy(column)"
+            v-for="(column, index) in fields"
+            :key="index"
+            scope="col"
+            :class="isColumnSort(column) ? 'cursor' : ''"
+            :style="getWidthStyle(column)"
+          >
+            <slot :name="`head(${column.key})`">
+              {{ column.label }}
+            </slot>
+            <font-awesome-icon
+              v-if="column.tooltip"
+              icon="info-circle"
+              class="mr-1"
+              v-b-popover.hover.top="column.tooltip"
+            />
+            <font-awesome-icon
+              v-if="column.key === sortBy"
+              :icon="descOrder ? 'caret-down' : 'caret-up'"
+              :class="darkMode ? 'text-white' : 'text-primary'"
+            />
+          </th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr v-for="pool in paginatedItems" :key="pool.id">
```

Comment:
Still using ```pool``` as a name, lets rename this to something more generic.

Reply:
Also it's being assumed that ```id``` is being passed down, if you're requiring this then lets update the ```items``` prop from ```any``` to something like

```
export interface Item {
  id: string;
  [string: string]: any;
}

```

Reply:
@Velua can you elaborate on this. Not sure what you mean

Reply:
done